docs/plugins/create-a-plugin.md

```diff
@@ -1,27 +1,45 @@
 ---
 id: create-a-plugin
-title: Create a Backstage Plugin
-description: Documentation on How to Create a Backstage Plugin
+title: Create a Frontend Plugin
+description: Documentation on How to Create a Backstage Frontend Plugin
 ---
 
-A Backstage Plugin adds functionality to Backstage.
+A Backstage Frontend Plugin adds functionality, via extensions, to the Backstage frontend web application.
 
-## Create a Plugin
+:::info
+To create a Backend Plugin, please see [Creating a Backend Plugin](./backend-plugin.md#creating-a-backend-plugin).
+:::
 
-To create a new frontend plugin, make sure you've run `yarn install` and installed
-dependencies, then run the following on your command line (a shortcut to
-invoking the
-[`backstage-cli new --select plugin`](../tooling/cli/03-commands.md#new))
+## Create a Frontend Plugin
+
+To create a new frontend plugin from our predefined template, first make sure you've run `yarn install`, then run the
+following on your command line (a shortcut to invoking the [`backstage-cli new --select plugin`](../tooling/cli/03-commands.md#new))
 from the root of your project.
 
-```bash
-yarn new --select plugin
+![Example of output when creating a new frontend plugin](../assets/getting-started/create-plugin_output.png)
+
+This will template out the files for a new Frontend Plugin into a folder based on the plugin ID that was provided.
+
+The `App.tsx` file will also be updated to include a route to the new generated demo page, like so:
+
+```tsx title=packages/app/src/App.tsx
+<FlatRoutes>
+  ...
+  <Route path="/my-plugin" element={<MyPluginPage />} />
+  ...
+</FlatRoutes>
 ```
 
-![Example of output when creating a new plugin](../assets/getting-started/create-plugin_output.png)
+:::info
+Important Advanced Note!
+
+Placing `<MyPluginPage />` into the application is what registers everything in your plugin with the Backstage
+application. Do not remove this unless you know what you are doing.
 
-This will create a new Backstage Plugin based on the ID that was provided. It
-will be built and added to the Backstage App automatically.
+By default, plugins are discovered when the Backstage `AppManager` traverses the React node tree when React starts up.
+When it finds a plugin extension in the visible React tree, it then "registers" the associated plugin and makes its apis
+available to `useApi()`.
+:::
```

Comment:
Would it not be better to just provide documentation for cases where you don't have a frontend component? 

The Badges plugin is one that comes to mind as it doesn't have a route at all: https://github.com/backstage/community-plugins/blob/main/workspaces/badges/plugins/badges/README.md

This is how we add it on the Demo site if you want to see a working example: https://github.com/backstage/demo/blob/master/packages/app/src/App.tsx#L54

Reply:
Are you suggesting to *not* explain how it works and just show examples?

Reply:
I was thinking both but that might not have been clear. I felt like this was worth a dedicated section/paragraph explaining along with examples for this specific scenario. Does that make more sense?

Reply:
Oh okay! Great, I misunderstood your use of the word "just' :) Updating now...